### PR TITLE
docs: merge migration pages [v6]

### PR DESCRIPTION
**Pull request by wolovim**
### What was wrong?

- [v6] condense migration guides into one page. mostly motivated by being able to set up redirects across v6 and v7 branches
- ~~v7 migration guide not yet included within v6 docs; i'd think it should be.~~ edit: oh, yeah, the broken links. will just include a nudge to try v7 with a link to the `latest` build.
- readthedocs supports redirects, so going forward:
   > /v7_migration.html -> /migration.html#migrating-from-v6-to-v7
   > /v6_migration.html -> /migration.html#migrating-from-v5-to-v6 
   > /v5_migration.html -> /migration.html#migrating-from-v4-to-v5 

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: newsfragments/3379.docs.rst
Comment (kclowes) [inline newsfragments/3379.docs.rst:1]: Are both of these newsfragments here on purpose?

**Review by kclowes — APPROVED** [review]
🚀

File: newsfragments/3379.docs.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Merge migration guides into one page
```
Comment (fselmo) [inline newsfragments/3379.docs.rst]: This needs to be done away with, yeah?

**Review by fselmo — APPROVED** [review]
lgtm, just need to remove the `v7` newsfragment

**Review by pacrob — APPROVED** [review]
lgtm, sans extra newsfragment :)

**Pull request merged**